### DISCUSSION TITLE
How to use cmds , executable file  not found in $PATH?

### DISCUSSION THREAD

**Discussion by masterwishx** [filebrowser/filebrowser#2019] [Q&A]
Using Docker for Unraid 
How to use command line ? i have executable not found for all commands ...

![image](https://user-images.githubusercontent.com/28630321/177051319-9ab54371-22fc-4ae3-a84e-ff26ae8bd45f.png)

**Comment by ramiresviana** [comment]
By default, File Browser executes the commands by calling their binaries directly. If you want to run them on a shell instead (such as Bash or PowerShell) you need to define it in `Settings -> Global Settings -> Execute on shell`

**Reply by exorcist-transfix-welsh** [reply]
Is there an example in the documentation of how this works? Can't for the life of me figure it out. Just getting 'Command not allowed.' even after adding it to the space separated list of allowed commands in settings.

**Comment by masterwishx** [comment]
Thanks for answer and cool Software.

Do you mean if i need to use command line i need enable (exec on shell) ?

can you please give example how i can use (how command should look) cmd with or without (Execute on shell) ?

should i also list all commands in user setting ?

using for now only (Command Runner)   =    After copy  (chmod 0777 $DESTINATION)   and etc ... 

for now only using

**Reply by ramiresviana** [reply]
Command Runner and Shell are different things, as described in the configuration:

> Global Settings > Command Runner
*Here** you can set commands that are executed in the named events. You must write one per line. The environment variables FILE and SCOPE will be available, being FILE relative to SCOPE. For more information about this feature and the available environment variables, please read the [documentation](https://filebrowser.org/configuration/command-runner).

> Global Settings > Execute on shell
By default, File Browser executes the commands by calling their binaries directly. If you want to run them on a shell instead (such as Bash or PowerShell), you can define it here with the required arguments and flags. If set, the command you execute will be appended as an argument. This apply to both user commands and event hooks.

> User Management > Commands
A space separated list with the available commands for this user. Example: git svn hg.

To allow users to run commands using the shell available on the interface, the list of commands must be provided on the User Management settings.

**Comment by rexzhang** [comment]
The official documentation is confusing, this is my settings and it works

URL: `/settings/global`
<img width="496" alt="截屏2023-02-14 11 08 07" src="https://user-images.githubusercontent.com/238707/218629083-3cc95be3-616a-42f5-9771-0a644b0a3e84.png">

URL `/settings/users/x`
<img width="496" alt="截屏2023-02-14 11 12 46" src="https://user-images.githubusercontent.com/238707/218629746-09c6604d-7131-4970-b4de-d26cad075ca1.png">
